src/main/cpp/src/api/genomicsdb_json_processor.cc

```diff
@@ -0,0 +1,196 @@
+  rapidjson::Document *json_doc = TO_JSON_DOCUMENT(m_json_document);
+  auto& allocator = json_doc->GetAllocator();
+  auto sample_info = m_sample_info.find(sample_name);
+  if (sample_info == m_sample_info.end()) {
+    // Create Value Arrays and add to map
+    std::vector<void *> fields;
+    rapidjson::Value *chrom = new rapidjson::Value(rapidjson::kArrayType);
+    fields.push_back(chrom);
+    if (m_payload_mode == all) {
+      rapidjson::Value *pos = new rapidjson::Value(rapidjson::kArrayType);
+      fields.push_back(pos);
+      for (auto field_name : m_field_names) {
+        rapidjson::Value *field_array = new rapidjson::Value(rapidjson::kArrayType);
+        fields.push_back(field_array);
+      }
+    }
+    m_sample_info.emplace(std::make_pair(sample_name, std::move(fields)));
+    sample_info = m_sample_info.find(sample_name);
+  }
+
+  auto i = 0u;
+  auto& fields = sample_info->second;
+
+  rapidjson::Value *chrom = reinterpret_cast<rapidjson::Value *>(fields[i++]);
+  rapidjson::Value chrom_value = rapidjson::Value(rapidjson::kStringType);
+  chrom_value.SetString(genomic_interval.contig_name.c_str(),
+                        genomic_interval.contig_name.length(),
+                        allocator);
+  chrom->PushBack(chrom_value, allocator);
+
+  if (m_payload_mode != all) return;
+
+  rapidjson::Value *pos = reinterpret_cast<rapidjson::Value *>(fields[i++]);
+  pos->PushBack(rapidjson::Value(genomic_interval.interval.first).Move(), allocator);
+
+  for (auto field_name : m_field_names) {
+    auto field_type = get_genomic_field_types()->at(field_name);
+    
+    bool found = false;
+    for (auto genomic_field: genomic_fields) {
+      if (genomic_field.name.compare(field_name) == 0) {
+        rapidjson::Value *values = reinterpret_cast<rapidjson::Value *>(fields[i++]);
+        if (STRING_FIELD(field_name, field_type)) {
+          rapidjson::Value value = rapidjson::Value(rapidjson::kStringType);
+          std::string str;
+          if (field_name == "GT") {
+            str = resolve_gt(genomic_fields);
+          } else {
+            str = genomic_field.to_string(field_type);
+          }
+          value.SetString(str.c_str(), str.length(), allocator);
+          values->PushBack(value, allocator);
+        } else if (INT_FIELD(field_type)) {
+          values->PushBack(rapidjson::Value(genomic_field.int_value_at(0)).Move(), allocator);
+        } else if (FLOAT_FIELD(field_type)) {
+          values->PushBack(rapidjson::Value(genomic_field.float_value_at(0)).Move(), allocator);
+        } else {
+          std::string msg = "Genomic field type for " + field_name + " not supported";
+          throw GenomicsDBException(msg.c_str());
+        }
+        found = true;
+        break;
+      }
+    }
+
+    if (!found) {
+      rapidjson::Value *values = reinterpret_cast<rapidjson::Value *>(fields[i++]);
+      rapidjson::Value value = rapidjson::Value(rapidjson::kStringType);
+      value.SetString(".", allocator);
```

Comment:
I know I went back and forth over this on slack...but it looks like json does have a null data type -- maybe we can use that? https://rapidjson.org/namespacerapidjson.html#ae79a4751c1c460ff0de5ecc07874f3e4

Reply:
I'm fine with the null value. At least as currently being used, the `.` won't help much.